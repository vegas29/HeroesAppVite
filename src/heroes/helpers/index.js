export * from './getHeroesByPublisher';
export * from './getHeroesByName';
export * from './getHeroById';
// export * from './heroImages';