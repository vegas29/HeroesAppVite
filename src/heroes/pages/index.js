export * from './DCPage';
export * from './HeroePage';
export * from './MarvelPage';
export * from './SearchPage';
