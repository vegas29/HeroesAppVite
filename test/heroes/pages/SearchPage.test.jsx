import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { SearchPage } from "../../../src/heroes";

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate
}));

describe('Pruebas en <SearchPage />', () => {

    beforeEach(() => jest.clearAllMocks());

    test('Debe de mostrarse correctamente con valores por defecto', () => {
        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )

        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar a Batman y el input con el valor del queryString', () => {

        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('batman');

        const img = screen.getByRole('img');
        expect( img.src ).toContain('assets/img/dc-batman.jpg');

    });


    test('Debe de mostrar un error si no se encuentra el hero (batman123)', () => {
        
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        );

        screen.debug();

        const alert = screen.getByLabelText('alert-danger');
        expect( alert.style.textAlign ).toBe('');
        
    });

    test('Debe de llamar el navigate a la pantalla nueva', () => {
        
        const inputValue = 'superman';

        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change( input, { target: { name: 'searchText', value: inputValue }})
        
        
        screen.debug();
        const form = screen.getByLabelText('form');
        fireEvent.submit( form );
        
        expect( mockedUseNavigate ).toHaveBeenCalledWith(`?q=${ inputValue }`)

    });

});