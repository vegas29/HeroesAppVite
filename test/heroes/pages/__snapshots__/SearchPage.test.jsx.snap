// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1
    class="text-center font-bold mt-10 text-2xl"
  >
     Busquedas 
  </h1>
  <div
    class="bg-white shadow-md rounded-md"
  >
    <div
      class="text-center py-7 md:py-10 md:w-5/12 mx-auto px-5"
    >
      <form
        aria-label="form"
      >
        <span
          class="font-semibold text-md my-2 block"
        >
          Inserte el nombre del héroe
        </span>
        <input
          autocomplete="false"
          class="w-full px-5 rounded-md shadow-sm h-12 bg-gray-100"
          name="searchText"
          placeholder="Buscar un héroe"
          type="text"
          value=""
        />
        <input
          class="w-full  bg-blue-500 hover:bg-blue-600 text-white cursor-pointer mt-5 py-2 rounded-md text-md font-semibold"
          type="submit"
          value="Buscar"
        />
      </form>
    </div>
  </div>
  <h2
    class="text-center font-bold mt-10 text-2xl"
  >
    Resultados
  </h2>
  <div
    class="bg-white text-gray-400 text-center text-2xl font-bold shadow-lg rounded-lg w-full md:w-5/12 mx-auto py-10 mt-5"
  >
    Busca un heroe, para que se muestre aquí
  </div>
  <div
    class="text-center py-7 md:py-10 md:max-w-7xl mx-auto px-5 grid grid-cols-1 md:grid-cols-3 gap-5"
  />
</div>
`;
